lib/puppet/type/file/content.rb

```diff
@@ -122,9 +122,8 @@ def retrieve
 
     # Make sure we're also managing the checksum property.
     def should=(value)
-      # treat the value as a bytestring, in Ruby versions that support it, regardless of the encoding
-      # in which it has been supplied
-      value = value.dup.force_encoding(Encoding::ASCII_8BIT) if value.respond_to?(:force_encoding)
+      # treat the value as a bytestring
+      value = value.b if value.is_a?(String)
```

Comment:
`.b` added in [Ruby 2.0](https://ruby-doc.org/core-2.0.0/String.html#method-i-b) 👍